cmd/signature-v2.go

```diff
@@ -157,6 +157,11 @@ func doesPresignV2SignatureMatch(r *http.Request) APIErrorCode {
 		return ErrExpiredPresignRequest
 	}
 
+	encodedResource, err = getResource(encodedResource, r.Host, globalDomainName)
+	if err != nil {
+		return ErrInternalError
```

Comment:
getResource can return APIErrorCode here @krishnasrinivas so that we don't need to mask the error here, we can return back proper errors..

Reply:
We don't have a suitable APIErrorCode for this and hence returned the generic internal error. Did not introduce a new error as this would be a corner case. You think we should introduce a new APIErrorCode?

Reply:
Was thinking we can return something like ErrInvalidArgument ..

Reply:
Added a more generic `ErrInvalidRequest` (http.StatusBadRequest) which should be judiciously used.